test/jasmine/tests/validate_test.js

```diff
@@ -440,4 +440,69 @@ describe('Plotly.validate', function() {
             'In layout, key yaxis2.overlaying (set to \'x\') got reset to \'false\' during defaults.'
         );
     });
+
+    it('catches bad axes in grid definitions', function() {
+        var out = Plotly.validate([
+            {y: [1, 2]},
+            {y: [1, 2], xaxis: 'x2', yaxis: 'y2'}
+        ], {
+            grid: {xaxes: ['x3', '', 'x2', 4], yaxes: ['y', 3, '', 'y4']},
+            // while we're at it check on another info_array
+            xaxis: {range: [5, 'lots']}
+        });
+
+        expect(out.length).toBe(5);
+        assertErrorContent(
+            out[0], 'dynamic', 'layout', null,
+            ['grid', 'xaxes', 0], 'grid.xaxes[0]',
+            'In layout, key grid.xaxes[0] (set to \'x3\') got reset to \'\' during defaults.'
+        );
+        assertErrorContent(
+            out[1], 'value', 'layout', null,
+            ['grid', 'xaxes', 3], 'grid.xaxes[3]',
+            'In layout, key grid.xaxes[3] is set to an invalid value (4)'
+        );
+        assertErrorContent(
+            out[2], 'value', 'layout', null,
+            ['grid', 'yaxes', 1], 'grid.yaxes[1]',
+            'In layout, key grid.yaxes[1] is set to an invalid value (3)'
+        );
+        assertErrorContent(
+            out[3], 'dynamic', 'layout', null,
+            ['grid', 'yaxes', 3], 'grid.yaxes[3]',
+            'In layout, key grid.yaxes[3] (set to \'y4\') got reset to \'\' during defaults.'
+        );
+        assertErrorContent(
+            out[4], 'dynamic', 'layout', null,
+            ['xaxis', 'range', 1], 'xaxis.range[1]',
+            'In layout, key xaxis.range[1] (set to \'lots\') got reset to \'50\' during defaults.'
```

Comment:
Hmm. Why does `xaxis.range[1]` get reset to `'50'`? here? I thought the `xaxis.range[1]` default was 6?

Reply:
It gets `range[0] * 10` https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/set_convert.js#L324

Reply:
ok thanks :ok_hand: